src/main/java/org/apache/maven/plugins/javadoc/AbstractJavadocMojo.java

```diff
@@ -334,9 +334,11 @@ public abstract class AbstractJavadocMojo extends AbstractMojo {
 
     /**
      * Specify if the Javadoc should operate in offline mode.
+     *
+     * @since 3.6.0
      */
-    @Parameter(defaultValue = "${settings.offline}", required = true, readonly = true)
-    private boolean isOffline;
+    @Parameter(property = "maven.javadoc.offline", defaultValue = "false")
+    private boolean offline;
```

Comment:
We can add information that Maven offline mode `-or, --offline` is also respected and has a higher priority

Reply:
My personal assumption for all maven plugins is that every plugin implicitly respects it (and I am allowed to get upset if they don't) because it is a tool wide facility).

I added some comments to this option and the one that it affects (`<links>`).